translatepy/translators/base.py

```diff
@@ -548,7 +551,13 @@ def _detect_and_validate_lang(self, language: str) -> str:
         else:
             result = Language(language)
 
-        return self._language_normalize(result)
+        normalized_result = self._language_normalize(result)
+
+        if self._supported_languages:  # Check if the attribute is not empty
+            if normalized_result not in self._supported_languages:
+                raise UnsupportedLanguage("Language {language_code} doesn't supported by {service}".format(language_code=language, service=str(self)))
```

Comment:
Typo: It should be `The language {language_code} is not supported by {service}`

Reply:
Ok